hack/versions/pkg/schema/git.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schema
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/pkg/errors"
+)
+
+type gitClient interface {
+	getChangedFiles() ([]string, error)
+	getFileFromRef(path string, ref string) ([]byte, error)
+	diffWithRef(path string, ref string) ([]byte, error)
+}
+
+type git struct {
+	path string
+}
+
+func newGit() (gitClient, error) {
+	gitPath, err := exec.LookPath("git")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find git on PATH")
+	}
+	return &git{
+		path: gitPath,
+	}, nil
+}
+
+func (g *git) getChangedFiles() ([]string, error) {
```

Comment:
Shd we just check if `pkg/skaffold/schema/latest/config.go` changes?

```
find pkg/skaffold/schema -name 'config.go'
pkg/skaffold/schema/v1beta1/config.go
pkg/skaffold/schema/v1beta8/config.go
pkg/skaffold/schema/v1beta7/config.go
pkg/skaffold/schema/v1beta10/config.go
pkg/skaffold/schema/v1beta9/config.go
pkg/skaffold/schema/v1beta2/config.go
pkg/skaffold/schema/v1alpha1/config.go
pkg/skaffold/schema/v1alpha5/config.go
pkg/skaffold/schema/v1alpha3/config.go
pkg/skaffold/schema/v1alpha4/config.go
pkg/skaffold/schema/v1alpha2/config.go
pkg/skaffold/schema/v1beta11/config.go
pkg/skaffold/schema/v1beta6/config.go
pkg/skaffold/schema/v1beta3/config.go
pkg/skaffold/schema/v1beta12/config.go
pkg/skaffold/schema/v1beta13/config.go
pkg/skaffold/schema/v1beta5/config.go
pkg/skaffold/schema/v1beta4/config.go
pkg/skaffold/schema/latest/config.go
```
Ideally, a contributor should never change a released config.
It will just make the script a little more efficient.